### PR TITLE
feat: add unit tests

### PR DESCRIPTION
**Pull request by gee4vee**

### CODE REVIEW

File: examples/java-spring-app.yaml [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 metadata:
-  name: java-spring app
+  name: java-spring-app
 spec:
   templateRepo:
     templateOwner: IBM
     templateRepoName: java-spring-app
     name: java-spring-app
-    owner: <OWNER>
+    owner: gee4vee
```
Comment (jmeis) [inline examples/java-spring-app.yaml:10]: Can you revert this? You can keep the name updated.

File: go.mod [hunk 2]
```diff
@@ -1,12 +1,23 @@
 go 1.13
 
 require (
+	github.com/go-delve/delve v1.5.0 // indirect
```
Comment (jmeis) [inline go.mod:4]: Do we need this dependency?

File: pkg/controller/starterkit/starterkit_controller.go [hunk 3]
```diff
@@ -30,6 +31,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
+	"github.com/go-logr/logr"
```
Comment (jmeis) [inline pkg/controller/starterkit/starterkit_controller.go:34]: I think we already have a logger, and don't need this one.

File: go.mod [hunk 4]
```diff
@@ -1,12 +1,23 @@
...
+	github.com/go-logr/logr v0.1.0
```
Comment (jmeis) [inline go.mod:5]: Do we need this dependency?

File: pkg/controller/starterkit/starterkit_controller.go [hunk 5]
```diff
@@ -418,9 +436,82 @@ func (r *ReconcileStarterKit) Reconcile(request reconcile.Request) (reconcile.Re
...
+	// Update CR
+	err := r.client.Update(context.TODO(), s)
+	if err != nil {
+		reqLogger.Error(err, "Failed to update StarterKit with finalizer")
+		return err
+	}
+	return nil
+}
+
+func (r *ReconcileStarterKit) finalizeStarterKit(reqLogger logr.Logger, request reconcile.Request, s *devxv1alpha1.StarterKit, githubClient *github.Client) error {
```
Comment (jmeis) [inline pkg/controller/starterkit/starterkit_controller.go:497]: Could you add a comment above this explaining what it does?

File: pkg/controller/starterkit/starterkit_controller.go [hunk 6]
```diff
@@ -418,9 +436,82 @@ func (r *ReconcileStarterKit) Reconcile(request reconcile.Request) (reconcile.Re
...
+func contains(list []string, s string) bool {
+	for _, v := range list {
+		if v == s {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *ReconcileStarterKit) addFinalizer(reqLogger logr.Logger, s *devxv1alpha1.StarterKit) error {
```
Comment (jmeis) [inline pkg/controller/starterkit/starterkit_controller.go:484]: Could you add a comment above this explaining what it does?

File: pkg/controller/starterkit/starterkit_controller.go [hunk 7]
```diff
@@ -418,9 +436,82 @@ func (r *ReconcileStarterKit) Reconcile(request reconcile.Request) (reconcile.Re
...
+	reqLogger.Info("Adding Finalizer for the StarterKit")
+	controllerutil.AddFinalizer(s, starterkitFinalizer)
+
+	// Update CR
+	err := r.client.Update(context.TODO(), s)
```
Comment (jmeis) [inline pkg/controller/starterkit/starterkit_controller.go:489]: Can you update the context?

File: Makefile [hunk 8]
```diff
@@ -0,0 +1,85 @@
+.DEFAULT_GOAL:=help
```
Comment (jmeis) [inline Makefile:1]: Could you update the dev guide with how to use the *Makefile*?

File: go.mod (hunk 4 above)
Comment (gee4vee) [reply go.mod:5]: The `logr` one gets added with `go mod tidy` so I think it's required.

File: pkg/controller/starterkit/starterkit_controller.go [hunk 9]
```diff
@@ -418,9 +436,82 @@ func (r *ReconcileStarterKit) Reconcile(request reconcile.Request) (reconcile.Re
...
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+		return reconcile.Result{}, nil
+	}
+
+	// Add finalizer for this CR
+	if !contains(instance.GetFinalizers(), starterkitFinalizer) {
+		reqLogger.Info("Adding finalizer to starterkit")
```
Comment (jmeis) [inline pkg/controller/starterkit/starterkit_controller.go:464]: There are a few different spelling/capitalizations for "starter kit" in the logs. Could you make that more consistent?

File: pkg/controller/starterkit/starterkit_controller.go [hunk 10]
```diff
@@ -418,9 +436,82 @@ func (r *ReconcileStarterKit) Reconcile(request reconcile.Request) (reconcile.Re
...
+			return reconcile.Result{}, err
+		}
+	} else {
+		reqLogger.Info("Starterkit already has finalizer")
+	}
+
 	return reconcile.Result{}, nil
 }
 
+func contains(list []string, s string) bool {
```
Comment (jmeis) [inline pkg/controller/starterkit/starterkit_controller.go:475]: Not sure if we should have a utils file and move all of these helper functions there.

File: Makefile (hunk 8 above)
Comment (gee4vee) [reply Makefile:1]: I added a note about using `make install-local`, but the rest of the Makefile only applies for deploying the operator to the cluster. I still haven't gotten that far.

File: pkg/controller/starterkit/starterkit_controller.go (hunk 10 above)
Comment (gee4vee) [reply pkg/controller/starterkit/starterkit_controller.go:475]: I've leave that for the code cleanup issue #720 we have.

File: examples/java-spring-app.yaml (hunk 1 above)
Comment (gee4vee) [reply examples/java-spring-app.yaml]: Fixed

File: examples/java-spring-app.yaml [hunk 11]
```diff
@@ -10,7 +10,7 @@ spec:
     owner: <OWNER>
     repoDescription: example code pattern
     secretKeyRef:
-      name: <NAME>
-      key: <KEY>
+      name: gabe-skit-operator
```
Comment (jmeis) [inline examples/java-spring-app.yaml:13]: could you revert this too?

**Pull request merged**